### PR TITLE
Feature/built in navigation reducer

### PR DESCRIPTION
**Pull request by j8seangel**
Since we have the `withNavigation` HOC available, make our own reducer doesn't add any advantage so this PR moves the reducer logic to the built-in reducer in the library and now we always navigate with the [library props](https://reactnavigation.org/docs/navigators/navigation-prop).

To handle the first state there is a new [home component](https://github.com/Vizzuality/forest-watcher/pull/25/files#diff-0efba47703bc875d8908055e75fc72a2R11) that checks if the setup was done or should redirect to the dashboard directly. 

Also saves in the AsyncStorage the setup as complete [here](https://github.com/Vizzuality/forest-watcher/compare/feature/built-in-navigation-reducer?expand=1#diff-8f5a241929ad13a2accef2b2a6169643R21)

### CODE REVIEW

File: app/components/home/index.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  async checkSetup() {
+    this.setup = true;
+    const userToken = await getToken();
+    const setupStatus = await getSetupStatus();
+    this.props.setLoginStatus({
+      loggedIn: true,
+      token: userToken
+    });
+
+    this.props.navigateReset(setupStatus ? 'Dashboard' : 'Setup');
```
Comment (geriux) [inline app/components/home/index.js]: Niceee!

File: app/redux-modules/navigation.js [hunk 2]
```diff
@@ -3,23 +3,32 @@ import { AppNavigator } from 'app/main';
 // Actions
```
Comment (geriux) [inline app/redux-modules/navigation.js]: Let's not forget to delete this

Comment (j8seangel) [inline app/redux-modules/navigation.js]: Yes, thanks!

**Comment by j8seangel** [comment]
I've included a link in the description to understand what an [HOC](https://facebook.github.io/react/docs/higher-order-components.html) is.

**Pull request merged**